### PR TITLE
[Java SDK core] emit watermark from PeriodicSequence (#23301)

### LINKED ISSUE #23301: [Bug]: Java SDK: PeriodicSequence must update watermark using WatermarkEstimator
### What happened?

https://lists.apache.org/thread/m2x78d0kmzb7y4frt052qctt4y5s2pzx

### Issue Priority

Priority: 2

### Issue Component

Component: sdk-java-core

### PR DESCRIPTION
**Pull request by je-ik**
Closes #23301

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by je-ik** [comment]
R: @damccorm

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PeriodicSequence.java [hunk 1]
```diff
@@ -178,9 +182,22 @@ public RestrictionTracker<OffsetRange, Long> newTracker(@Restriction OffsetRange
       return new OutputRangeTracker(restriction);
     }
 
+    @GetInitialWatermarkEstimatorState
+    public Instant getInitialWatermarkState() {
+      return BoundedWindow.TIMESTAMP_MIN_VALUE;
```
Comment (damccorm) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PeriodicSequence.java:187]: Should we use the restriction's minimum value here? (or maybe slightly less than that to allow the first element to be emitted)

**Review by je-ik — COMMENTED** [review]
I'll make sure the checks are green, can we merge this then?

**Comment by je-ik** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by je-ik** [comment]
Run Java_Examples_Dataflow_Java17 PreCommit

**Comment by je-ik** [comment]
Run Java_Examples_Dataflow_Java11 PreCommit

**Review by damccorm — APPROVED** [review]
SGTM

**Pull request merged**